requirements/requirements_nlp.txt

```diff
@@ -8,6 +8,7 @@ gdown
 h5py
 ijson
 jieba
+llama-index==0.10.43
```

Comment:
llama index is an annoyingly heavy dependency - why is it being added as a hard dependency? 
https://github.com/run-llama/llama_index/blob/main/pyproject.toml#L52-L63
Each of these has a nest of dependencies on its own. 

@ericharper  what exactly uses this and why is it necessary to be a hard dependency?